test-requirements.txt

```diff
@@ -5,7 +5,7 @@ bashate>=0.2 # Apache-2.0
 flake8<2.6.0,>=2.5.4 # MIT
 pyasn1 # BSD
 pyOpenSSL>=0.14 # Apache-2.0
-requests!=2.12.2,!=2.13.0,>=2.10.0 # Apache-2.0
+requests[security]
```

Comment:
@tonytan4ever I don't see us installing this package anywhere else, can you document in the commit message why this is needed? Also, what is the implication of removing the pin for requests that was previously there?

Reply:
This is for fixing the error:

```UPPER_CONSTRAINTS_PROTO=http
++ export 'UPPER_CONSTRAINTS_FILE=https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=master'
++ UPPER_CONSTRAINTS_FILE='https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=master'
++ pip install --requirement requirements.txt --constraint 'https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt?h=master'
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/usr/local/lib/python2.7/dist-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Certificate did not match expected hostname: git.openstack.org. Certificate: {'subject': ((('organizationalUnitName', u'Domain Control Validated'),), (('organizationalUnitName', u'PositiveSSL'),), (('commonName', u'git.airshipit.org'),)), 'subjectAltName': (('DNS', 'git.airshipit.org'), ('DNS', 'www.git.airshipit.org')), 'notAfter': 'May 19 23:59:59 2019 GMT'}
Exception:```

Reply:
`UPPER_CONSTRAINTS_FILE` points to http:// and not https:// as per your snippet, which I believe is what fixed this issue. I'd double-check if this is actually needed.